src/librustc_target/spec/x86_64_apple_darwin.rs

```diff
@@ -6,6 +6,12 @@ pub fn target() -> TargetResult {
     base.max_atomic_width = Some(128); // core2 support cmpxchg16b
     base.eliminate_frame_pointer = false;
     base.pre_link_args.insert(LinkerFlavor::Gcc, vec!["-m64".to_string()]);
+    if let Some(sysroot) = super::apple_base::sysroot("macosx")? {
+        base.pre_link_args.insert(
+            LinkerFlavor::Gcc,
+            vec!["-isysroot".to_string(), sysroot.clone(), "-Wl,-syslibroot".to_string(), sysroot],
```

Comment:
Do you know what the expected impact will be from setting these flags? I'm not sure if this is likely to cause issues or perhaps cause misconfigurations on some systems.